### ISSUE TITLE
Variable to get a list of results of a task

### ISSUE THREAD

**Issue by hokadiri** [tektoncd/pipeline#4097]
### Feature request
if a i have a list of results

`apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: some-task
spec:
  results:
    - name: A
    - name: B
    - name: C
    - name: D
    - name: E
    - name: F
`
I want a [variable](https://tekton.dev/docs/pipelines/variables/) that tells me the list so i can loop through it in in a task shell
`
      script: |
        # loop through $(results.list)
`

A variable which lists the results defined in a task

### Use case
I have a list of 10 results , i want to set them to some value.
Right now i have to 
`
echo "X" > "$(results.A.path)" 
echo "Y" > "$(results.B.path)" 
echo "Z" > "$(results.C.path)" 
...
`

Would be great if i can loop through the results list/

Supports DRY principle.

**Comment by bobcatfish** [comment]
Thanks for opening this @hokadiri !

Are you able to share any more details around the use case you need this for? e.g. a specific Task with specific results would really help us understand why this is needed.

(I'm also wondering if in this case you'd prefer to be able to have an array or dictionary type result - but I think even if we supported those, we've been hesitant to support variable replacement for going from an array to a string b/c it's hard to say how it should be formatted - e.g. separated by commas? spaces? something else?)

**Comment by hokadiri** [comment]
Hi @bobcatfish, I have a tekton task that runs a tool which takes a git commit and outputs the list of tests to run with some additional metadata
```
{
  "test-a": {
    "cmd": "some cmd",
     ... some other metadata
  },
  "test-c": {
    "cmd": "some cmd",
    ... some other metadata
  },
  "test-d": {
    "cmd": "some cmd"
    ... some other metadata
  },
  "test-e": {
    "cmd": "some cmd"
    ... some other metadata
  },
}
```

Ideally I want to take this json and trigger concurrent task runs (One test per task).

Since results can't be lists/dictonaries, workaround is i'm creating a result for each test, looping thru and setting it.

To set it, I wanted to loop through the result list
```
For each test in $(results.list)
  if test in json:
    set result.test to true
  else:
    set result.test to false 
```

Ideally if results can support multiple types (not just string), then I can do:
run task to generate json output -> convert json to results dict -> pass results dict to tekton taskB iterparams -> task-loop concurrently -> fan-in -> done

**Comment by bobcatfish** [comment]
ooooo thanks for the additional info!! i've actually just created 2 proposals that might help you out:

* [TEP-0076 array results](https://github.com/tektoncd/community/pull/477)
* [TEP-0075 dictionary params + reseults](https://github.com/tektoncd/community/pull/479)

(Eventually I think we should allow even more complex types, e.g. arrays of dictionaries, etc.)

Maybe even just having array results support would be enough to unblock you for now? (it sounds like [task-loops](https://github.com/tektoncd/experimental/tree/main/task-loops) might be relevant to your interests also, if you wanted to run the same Task "for each" test)

**Comment by hokadiri** [comment]
@bobcatfish , yeah having array results support - in conjunction with task-loops - would be enough to unblock me. 

In general being able to convert a json/yaml file to an equivalent object tekton understands and vice versa would be a very helpful feature.

**Comment by sbwsg** [comment]
/lifecycle frozen

Adding frozen lifecycle given its relevance to the [array results & indexing TEP](https://github.com/tektoncd/community/pull/477).
